internal/kafka/internal/services/data_plane_kafka.go

```diff
@@ -399,3 +469,15 @@ func buildRoutes(routesInRequest []dbapi.DataPlaneKafkaRouteRequest, kafka *dbap
 	}
 	return routes, nil
 }
+
+// getManagedKafkaDeploymentType gets the deployment type from the data plane kafka status
+// If the id of the status begins with "reserved" then the deployment type is reserved.
+// Otherwise, it is a "real" deployment
+func getManagedKafkaDeploymentType(ks *dbapi.DataPlaneKafkaStatus) managedKafkaDeploymentType {
+	parsedID := strings.Split(ks.KafkaClusterId, "-")
+	if parsedID[0] == reservedDeploymentType.String() {
+		return reservedDeploymentType
+	}
+
+	return realDeploymentType
```

Comment:
I think this would be more readable this way
```suggestion
	if (strings.HasPrefix(ks. KafkaClusterId,  reservedDeploymentType.String())) {
	    return reservedDeploymentType
	}

	return realDeploymentType
```

Reply:
Thanks. Updated to use the prefix check

Reply:
Has this been updated? I don't see it reflected

Reply:
let me double check, I might have forgotten to push

Reply:
Yes. Should be good now. Can you re-check? @miguelsorianod